ssz/sedes/base.py

```diff
@@ -0,0 +1,149 @@
+import abc
+
+from typing import (
+    Generic,
+    Tuple,
+    TypeVar,
+)
+
+from ssz.exceptions import (
+    DeserializationError,
+    SerializationError,
+)
+
+
+T = TypeVar("T")
+S = TypeVar("S")
```

Comment:
What are `T` and `S` standing for?

Reply:
`T` and `S` are used because they seem to be the standard names for type variables: https://docs.python.org/3/library/typing.html#user-defined-generic-types

My guess is that `T` stands for `Type` and `S` is just the letter that is closest to `T` .

Reply:
Sometimes, to give type vars more meaning we also name them like `TValue`, `TKey`, `TWhatever` as a convention borrowed from the .NET community ;)

Reply:
Hmm, I guess we could be more explicit here as well and get rid of the single letter variable. Maybe `TSerializable` and `TDeserialized`.

Reply:
This could still get cleaned up and I think more expressive names for these would improve readability.